extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/providers/KnownOidcProviders.java

```diff
@@ -19,13 +19,34 @@ public static OidcTenantConfig provider(OidcTenantConfig.Provider provider) {
             case LINKEDIN -> linkedIn();
             case MASTODON -> mastodon();
             case MICROSOFT -> microsoft();
+            case SLACK -> slack();
             case SPOTIFY -> spotify();
             case STRAVA -> strava();
             case TWITCH -> twitch();
             case TWITTER, X -> twitter();
         };
     }
 
+    private static OidcTenantConfig slack() {
+        OidcTenantConfig ret = new OidcTenantConfig();
+        ret.setAuthServerUrl("https://slack.com");
+        ret.setApplicationType(OidcTenantConfig.ApplicationType.WEB_APP);
+        ret.getCredentials().getClientSecret().setMethod(Method.POST);
+
+        var token = ret.getToken();
+        token.setPrincipalClaim("name");
+        token.setSubjectRequired(true);
+
+        var authentication = ret.getAuthentication();
+        authentication.setScopes(List.of("email", "profile"));
+        authentication.setForceRedirectHttpsScheme(true);
+        authentication.setAddOpenidScope(true);
+        authentication.setNonceRequired(true);
+        authentication.failOnMissingStateParam = true;
```

Comment:
That is not specific to Slack

Reply:
https://quarkus.io/guides/all-config#quarkus-oidc_quarkus-oidc-authentication-fail-on-missing-state-param says:

_Reporting HTTP 401 error is usually the right thing to do in such cases, it minimizes a risk of the browser redirect loop but also can identify problems in the way SPA or Quarkus application manage redirects. _

Default value is false. Why exactly is setting it to `true` wrong? I don't understand your comment, please provide more context.

I am trying to find the most secure configuration, not just "something that works".